### PR TITLE
Fix #1131(workaround): explicitly code `__stop` in `stop` and `restart` methods of `Arbiter` class

### LINKED ISSUE #1131: quit --waiting command hang up
I think there is a critical issue after upgrade to tornado>5 .
```
circusctl --endpoint=xxx quit --waiting 
```
will successfully stop the process but hang up in terminal. 
After trying to debug this, I find the quit action stuck in https://github.com/circus-tent/circus/blob/2bedab09145a4959ce02b93777dbffbce56edd8e/circus/client.py#L142 lead to timeout. 

To my surprise, `quit` without `--waitting` is all fine.  

This issue can simply reproduced by running `circusd examples/example2.ini` and then stop arbiter with:
```
circusctl --timeout 10000 --endpoint tcp://127.0.0.1:5555 quit --waiting
```

### PR DESCRIPTION
**Pull request by unkcpz**
Issue #1131  when `quit` or `restart` is called with
`waiting` option, `arbiter.stop` is running to stop all the watchers,
processes, however, somehow `arbiter.stop()` is a coroutine which return
the Future is never get finished. Therefore, callback
`_dispatch_callback_future` of `controller` never get run.

After lots of debug attempts I find two workarounds.

1. When stop watchers and calling `Watcher().kill_process` do not wait
   `tornado_sleep(0.1)` if process is still alive. This may cause some
   unexpected side effects.
2. Just explicitly implement `__stop()` method of `Arbiter` class in
   `stop()` and `restart()`. vvvery strange!!! Anyway, this doesn't
   affect the code logic but get issue solved, so adapt to
   workaround.

pinging @biozz @k4nar for inputs.

### CODE REVIEW

**Comment by ltalirz** [comment]
Is help needed with reviewing this PR?
If yes, @sphuber or @muhrin might be able to give a look as well

**Comment by sphuber** [comment]
Would love to pitch in, but really out of my depth here.

**Comment by sphuber** [comment]
@k4nar is there anything specific you would want to verify or test before we could merge this? If I understood correctly, this is what is blocking a `0.17.0` release with `tornado>=5` compatibility, which is quite important for a number of users of this package. I am not sure if I can really help with reviewing this PR, but if there is anything else I can do, let me know. Thanks!

**Review by k4nar — APPROVED** [review]
This could probably be refactored somehow, but let's roll with it for now :+1: .

**Pull request merged**
